src/Sentry/Internal/NoOpTransaction.cs

```diff
@@ -0,0 +1,85 @@
+namespace Sentry.Internal;
+
+/// <summary>
+/// Transaction class to use when we can't return null but a request to create a transaction couldn't be completed.
+/// </summary>
+internal class NoOpTransaction : NoOpSpan, ITransaction
```

Comment:
Do we return this if we're not sampling a transaction? I wonder because how would trace-id propagation work. ANd also because we return `new` on props below so that would mean a bunch of allocations in that case, that wouldnt' be used at all.

Reply:
The only place it gets used [is here](https://github.com/getsentry/sentry-dotnet/blob/db8319dffdd9b3385d3e8acdff364b561a90628a/src/Sentry/Internal/Hub.cs#L123-L129), when the instrumenter configured doesn't match that on the span context. I imagine Matt did this so that something non-breaking would be returned and we could avoid throwing exceptions.

```csharp
        var instrumenter = (context as SpanContext)?.Instrumenter;
        if (instrumenter != _options.Instrumenter)
        {
            _options.LogWarning(
                $"Attempted to start a transaction via {instrumenter} instrumentation when the SDK is" +
                $" configured for {_options.Instrumenter} instrumentation.  The transaction will not be created.");

            return NoOpTransaction.Instance;
        }
```

In this case, there won't be any transaction or trace propagation. I think would happen if you'd wired up your app to use OpenTelemetry and then you called `SentrySdk.StartTransaction`... so trying to mix Sentry and OpenTelemetry span tracing in the same app.